File: java/src/edu/berkeley/cs/util/Queue.java
package edu.berkeley.cs.util;

import java.util.Iterator;

public class Queue<T> implements Iterable<T> {
  private LinkedList<T> list;

  public Queue() {
    list = new LinkedList<>();
  }

  public Queue(LinkedList<T> list) {
    this.list = list;
  }

  public T peek() {
    return list.front();
  }

  public void enqueue(T value) {
    list.insertEnd(value);
  }

  public T dequeue() {
    return list.removeFront();
  }

  public int size() {
    return list.size();
  }

  public boolean isEmpty() {
    return list.isEmpty();
  }

  @Override
  public Iterator<T> iterator() {
    return list.iterator();
  }
}

File: python/src/util/queue.py
from python.src.util.linkedlist import LinkedList


class Queue:

    def __init__(self):
        self.list = LinkedList()

    def __init(self, value):
        self.list = value

    def peek(self):
        return self.list.front()

    def enqueue(self, value):
        self.list.insert_end(value)

    def dequeue(self):
        return self.list.remove_front()

    def size(self):
        return self.list.size()

    def is_empty(self):
        return self.list.is_empty()
